### PR TITLE
feat: add journeyProfile table

### PR DESCRIPTION
**Pull request by storyworks**
# Description

Add a user profile table for api-journeys. 

[Basecamp Todo](https://3.basecamp.com/3105655/buckets/31245893/todos/5783931883)

# How should this PR be QA Tested?

- Would need to run locally and comment out "AuthGuard", and manually pass userId (instead of useCurrentUserId) to test.
- Run seed.
- [ ] Able to make a user profile for userId without profile with `journeyProfileCreate`
- [ ] Able to retrieve a user profile via the userId with `getJourneyProfile`

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/api-journeys/src/app/modules/journeyProfile/journeyProfile.graphql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+type JourneyProfile @key(fields: "id") {
+  id: ID!
+  userId: ID!
+  acceptedTermsAt: DateTime
+}
+
+extend type Query {
+  getJourneyProfile: JourneyProfile
+}
```
Comment (csiyang) [inline apps/api-journeys/src/app/modules/journeyProfile/journeyProfile.graphql:9]: add the update mutation? Or were you planning on doing that in another PR?

**Comment by csiyang** [comment]
Also getting the circular structure error when trying to follow QA steps locally

File: apps/api-journeys/src/app/modules/journeyProfile/journeyProfile.graphql (hunk 1 above)
Comment (storyworks) [reply apps/api-journeys/src/app/modules/journeyProfile/journeyProfile.graphql:9]: Oops forgot! Will do.

File: apps/api-journeys/src/app/modules/journeyProfile/journeyProfile.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  async journeyProfileCreate(
+    @CurrentUserId() userId: string
+  ): Promise<JourneyProfile> {
+    const profile = await this.getJourneyProfile(userId)
+
+    // Create profile after accepting terms of service
+    if (profile == null) {
+      return await this.journeyProfileService.save({
+        userId,
+        acceptedTermsAt: new Date().toISOString()
```
Comment (csiyang) [inline apps/api-journeys/src/app/modules/journeyProfile/journeyProfile.resolver.ts:31]: when I created a profile, the acceptedTermsAt value was null

**Review by csiyang — CHANGES_REQUESTED** [review]
Also fix merge conflict in stage

File: apps/api-journeys/src/app/modules/journeyProfile/journeyProfile.resolver.ts (hunk 2 above)
Comment (storyworks) [reply apps/api-journeys/src/app/modules/journeyProfile/journeyProfile.resolver.ts:31]: Not sure why that'd be unless you already had a profile in the db with acceptedTermsAt set to null... Nicks frontend change can help debug this.

Comment (storyworks) [reply apps/api-journeys/src/app/modules/journeyProfile/journeyProfile.resolver.ts:31]: fixed!

**Review by csiyang — APPROVED** [review]

**Pull request merged**